src/Mono.Android/Mono.Android.csproj

```diff
@@ -372,4 +373,13 @@
       <ReferenceOutputAssembly>False</ReferenceOutputAssembly>
     </ProjectReference>
   </ItemGroup>
+  <Target Name="AfterBuild">
```

Comment:
Should we add `Inputs/Outputs` here? https://github.com/xamarin/xamarin-android/blob/master/Documentation/guides/MSBuildBestPractices.md#incremental-builds

The reason would be if this slows our build time, does this target run on every build?:
```
> msbuild src\Mono.Android\Mono.Android.csproj
> msbuild src\Mono.Android\Mono.Android.csproj /clp:performancesummary
```
Is this target taking up time on the second one?

Reply:
We should run on every build, if we are building Mono.Android we should check if we are breaking the API

Reply:
the first one:

Reply:
Right so we only should check if `$(TargetPath)` changes, and is newer than a stamp file or something.

Something like (_untested_):

```xml
<Target Name="AfterBuild"
    Inputs="$(TargetPath)"
    Outputs="$(IntermediateOutputPath)\CheckApiCompatibility.stamp">
<!-- ... -->
  <Touch Files="$(IntermediateOutputPath)\CheckApiCompatibility.stamp" AlwaysCreate="True" />
  <ItemGroup>
    <FileWrites Include="$(IntermediateOutputPath)\CheckApiCompatibility.stamp" />
  </ItemGroup>
</Target>
```

This should make the second call skip `AfterBuild` if the assembly didn't change.

When I build `Xamarin.Android.sln`, I rarely change `Mono.Android.csproj` when working on MSBuild-related stuff. This should save ~3 seconds from my dev-loop.

Reply:
got it, I will add it. thanks for the tip!

Reply:
@jonathanpeppers Awesome Hint!